src/festim/advection.py

```diff
@@ -0,0 +1,100 @@
+import ufl
+from dolfinx import fem
+
+from festim.helpers import Value, nmm_interpolate
+from festim.species import Species
+from festim.subdomain import VolumeSubdomain
+
+
+class AdvectionTerm:
+    """
+    Advection term class
+
+    args:
+        velocity: the velocity field or function
+        subdomain: the volume subdomain where the velocity is to be applied
+        species: the species to which the velocity field is acting on
+
+    attributes:
+        velocity: the velocity field or function
+        subdomain: the volume subdomain where the velocity is to be applied
+        species: the species to which the velocity field is acting on
+
+    """
+
+    velocity: fem.Function
+    subdomain: VolumeSubdomain
+    species: Species
+
+    def __init__(
+        self,
+        velocity: fem.Function,
+        subdomain: VolumeSubdomain,
+        species: Species,
+    ):
+        self.velocity = velocity
+        self.subdomain = subdomain
+        self.species = species
+
+    @property
+    def velocity(self):
+        return self._velocity
+
+    @velocity.setter
+    def velocity(self, value):
+        if value is None:
+            self._velocity = Value(value)
+        elif isinstance(
+            value,
+            fem.Function,
+        ):
+            self._velocity = Value(value)
+        elif callable(value):
+            args = value.__code__.co_varnames
+            if "t" in args:
+                self._velocity = Value(value)
+            else:
+                raise TypeError("Advection field can only be a function of time (t)")
```

Comment:
This will not test that it has only `t` but simply that `t` is in the arguments

Therefore `lambda t, arg2, arg3, arg4: 2 * t` would pass